SimCalorimetry/EcalSimProducers/src/EcalDigiProducer_Ph2.cc

```diff
@@ -88,22 +94,58 @@ EcalDigiProducer_Ph2::EcalDigiProducer_Ph2(const edm::ParameterSet& params, edm:
 
-      m_APDResponse(!m_apdSeparateDigi
-                        ? nullptr
-                        : std::make_unique<EBHitResponse_Ph2>(
-                              m_ParameterMap.get(), &m_EBShape, true, m_apdParameters.get(), &m_APDShape)),
+      m_componentDigiTag(params.getParameter<std::string>("componentDigiTag")),
+      m_componentParameters(
+          std::make_unique<ComponentSimParameterMap>(params.getParameter<bool>("componentAddToBarrel"),
+                                                     m_componentSeparateDigi,
+                                                     params.getParameter<double>("simHitToPhotoelectronsBarrel"),
+                                                     0,  // endcap parameters not needed
+                                                     params.getParameter<double>("photoelectronsToAnalogBarrel"),
+                                                     0,
+                                                     params.getParameter<double>("samplingFactor"),
+                                                     params.getParameter<double>("componentTimePhase"),
+                                                     m_readoutFrameSize,
+                                                     params.getParameter<int>("binOfMaximum"),
+                                                     params.getParameter<bool>("doPhotostatistics"),
+                                                     params.getParameter<bool>("syncPhase"))),
+
+      m_APDResponse(!m_apdSeparateDigi ? nullptr
+                                       : std::make_unique<EBHitResponse_Ph2>(m_ParameterMap.get(),
+                                                                             &m_EBShape,
+                                                                             true,
+                                                                             false,
+                                                                             m_apdParameters.get(),
+                                                                             &m_APDShape,
+                                                                             m_componentParameters.get(),
+                                                                             &m_ComponentShapes)),
+
+      m_ComponentResponse(!m_componentSeparateDigi
+                              ? nullptr
+                              : std::make_unique<EBHitResponse_Ph2>(
+                                    m_ParameterMap.get(),
+                                    &m_EBShape,
+                                    false,
+                                    true,
+                                    m_apdParameters.get(),
+                                    &m_APDShape,
+                                    m_componentParameters.get(),
+                                    &m_ComponentShapes)),  // check if that false is correct // TODO HERE JCH
```

Comment:
did you check if the false is correct?

Reply:
Yes, it's correct. I removed the comment.